src/app/content/practiceQuestions/utils.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { book } from '../../../test/mocks/archiveLoader';
+import { findArchiveTreeNodeById } from '../utils/archiveTreeUtils';
+import { stripIdVersion } from '../utils/idUtils';
+import { PracticeQuestionsSummary } from './types';
+import {
+  getNextPageWithPracticeQuestions,
+  getPracticeQuestionsLocationFilters,
+  pageHasPracticeQuestions,
+} from './utils';
+
+describe('pageHasPracticeQuestions', () => {
+  it('returns true', () => {
+    expect(pageHasPracticeQuestions('pageId', { countsPerSource: { pageId: 2, other: 0 } })).toEqual(true);
+  });
+
+  it('returns false', () => {
+    expect(pageHasPracticeQuestions('asdasd', { countsPerSource: { pageId: 2, other: 0 } })).toEqual(false);
+  });
+});
+
+describe('getPracticeQuestionsLocationFilters', () => {
+  it('returns empty map if book or summary is not truthy', () => {
+    expect(getPracticeQuestionsLocationFilters(null, {} as any)).toEqual(new Map());
+    expect(getPracticeQuestionsLocationFilters({} as any, undefined)).toEqual(new Map());
+  });
+
+  it('returns location filters for practice questions', () => {
+    const page1 = (book as any).tree.contents[1].contents[0].contents[0];
+    const linkedPage1 = findArchiveTreeNodeById(book.tree, page1.id);
+    const page2 = (book as any).tree.contents[1].contents[1];
+    const linkedPage2 = findArchiveTreeNodeById(book.tree, page2.id);
+    const page3 = (book as any).tree.contents[1].contents[2];
+    const linkedPage3 = findArchiveTreeNodeById(book.tree, page3.id);
+    const page4 = (book as any).tree.contents[2].contents[0];
+    const linkedPage4 = findArchiveTreeNodeById(book.tree, page4.id);
```

Comment:
Could this be replaced by `findArchiveTreeNode(book.tree, hardcoded-page-id)`? The structure is hardcoded anyway and I'm not sure if it makes it more readable what belongs to where.